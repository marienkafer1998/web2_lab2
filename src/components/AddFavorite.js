import React from "react";


export default function AddFavorite(props) {
    return (
        <form  onSubmit={(e) => props.onSubmit(e)}>
            <input type="text" name="cityName" placeholder="City" required={true} />
            <input type="submit" value="Add to favorite"/>
        </form>
    );
}

