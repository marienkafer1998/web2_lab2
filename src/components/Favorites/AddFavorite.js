import React from "react";
import "./Favorites.css";
import Button from "reactstrap/es/Button";



export default function AddFavorite(props) {
    return (
        <form  onSubmit={(e) => props.onSubmit(e)}>
            <input  text-transform="capitalize" className={"adder"} type="text" name="cityName" placeholder="Add new city" required={true} />
            <Button type="submit" >+</Button>
        </form>
    );
}

