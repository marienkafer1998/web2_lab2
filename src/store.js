import { createStore, combineReducers, applyMiddleware } from "redux";
import logger from "redux-logger";
import thunk from "redux-thunk";
import { LOCAL_STORAGE_KEY } from "./LocalStorage";

import geoReducer from "./redux/geoReducer";
import favReducer from "./redux/favReducer";

const store = createStore(
    combineReducers({geo: geoReducer, fav: favReducer}),
    applyMiddleware(logger, thunk)
);



store.subscribe(() => {
    localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify([...store.getState().fav.favorites.keys()]));
});

export default store;
