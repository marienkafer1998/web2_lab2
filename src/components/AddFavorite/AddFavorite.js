import React from "react";
import "../Favorites/Favorites.css";
import {
    Button
} from 'reactstrap';



export default function AddFavorite(props) {
    return (
        <form  onSubmit={(e) => props.onSubmit(e)}>
            <input  text-transform="capitalize" className={"adder"} type="text" name="cityName" placeholder="Add new city" required={true} />
            <Button type="submit" >+</Button>
        </form>
    );
}

