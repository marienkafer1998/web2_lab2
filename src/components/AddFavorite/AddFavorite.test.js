import React from 'react';
import renderer from 'react-test-renderer';
import AddFavorite from "./AddFavorite.js";
import {Provider} from "react-redux";
import store from "../../store";

it('AddFavorite functional rendered properly', () => {
    const tree = renderer.create(
        <Provider store={store}>
            <AddFavorite/>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});


