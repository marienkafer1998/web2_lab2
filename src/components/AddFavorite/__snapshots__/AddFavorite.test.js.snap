// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFavorite functional rendered properly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="adder"
    name="cityName"
    placeholder="Add new city"
    required={true}
    text-transform="capitalize"
    type="text"
  />
  <button
    aria-label={null}
    className="btn btn-secondary"
    onClick={[Function]}
    type="submit"
  >
    +
  </button>
</form>
`;

exports[`AddFavorite functional renders  1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="adder"
    name="cityName"
    placeholder="Add new city"
    required={true}
    text-transform="capitalize"
    type="text"
  />
  <button
    aria-label={null}
    className="btn btn-secondary"
    onClick={[Function]}
    type="submit"
  >
    +
  </button>
</form>
`;

exports[`AddFavorite functional renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="adder"
    name="cityName"
    placeholder="Add new city"
    required={true}
    text-transform="capitalize"
    type="text"
  />
  <button
    aria-label={null}
    className="btn btn-secondary"
    onClick={[Function]}
    type="submit"
  >
    +
  </button>
</form>
`;

exports[`WeatherHere renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="adder"
    name="cityName"
    placeholder="Add new city"
    required={true}
    text-transform="capitalize"
    type="text"
  />
  <button
    aria-label={null}
    className="btn btn-secondary"
    onClick={[Function]}
    type="submit"
  >
    +
  </button>
</form>
`;
